test/cli_images_test.go

```diff
@@ -81,6 +84,36 @@ func (suite *PouchImagesSuite) TestImagesWorks(c *check.C) {
 	}
 }
 
+//TestImageListFilter test the filter flag works right
+func (suite *PouchImagesSuite) TestImageListFilter(c *check.C) {
+	busyBoxImageInfo, err := getImageInfo(apiClient, busyboxImage)
+	c.Assert(err, check.IsNil)
+
+	//Test Reference filter
+	referenceRes := command.PouchRun("images", "-f", "reference="+busyboxImage)
+	items := imagesListToKV(referenceRes.Combined())
+	c.Assert(len(items), check.Equals, 1)
+	c.Assert(items[busyboxImage][1], check.Equals, busyBoxImageInfo.RepoTags[0])
+
+	//Test before filter
+	beforeRes1 := command.PouchRun("images", "-f", "before="+busyboxImage)
+	items1 := imagesListToKV(beforeRes1.Combined())
+	beforeRes2 := command.PouchRun("images", "-f", "before="+helloworldImage)
+	items2 := imagesListToKV(beforeRes2.Combined())
+	c.Assert(len(items1)+len(items2), check.Equals, 1)
+
+	//Test since filter
+	sinceRes1 := command.PouchRun("images", "-f", "since="+busyboxImage)
+	items1 = imagesListToKV(sinceRes1.Combined())
+	sinceRes2 := command.PouchRun("images", "-f", "since="+helloworldImage)
+	items2 = imagesListToKV(sinceRes2.Combined())
+	c.Assert(len(items1)+len(items2), check.Equals, 1)
+
+	//Test invalid filter
+	invalidRes := command.PouchRun("images", "-f", "after="+busyboxImage)
+	c.Assert(invalidRes.Stderr(), check.Equals, "Error: failed to get image list: {\"message\":\"Invalid filter after\"}\n\n")
```

Comment:
the case is failed 

```
/home/travis/gopath/src/github.com/alibaba/pouch/test/cli_images_test.go:114:
    c.Assert(invalidRes.Stderr(), check.Equals, "Error: failed to get image list: {\"message\":\"Invalid filter after\"}\n\n")
... obtained string = "" +
...     "Error: failed to get image list: {\"message\":\"invalid filter after\"}\n" +
...     "\n"
... expected string = "" +
...     "Error: failed to get image list: {\"message\":\"Invalid filter after\"}\n" +
...     "\n"

```

I was thinking that we can do this in api level test and only check the status. The message isn't out of control.
But the status should be stable. WDYT?

Reply:
agree, update it later.